src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -93,14 +94,27 @@
     
     @EJB
     SystemConfig systemConfig;
+    
+    @Inject
+    DataverseSession session;
 
     private static final SimpleDateFormat logFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH-mm-ss");
     
-    @PersistenceContext(unitName = "VDCNet-ejbPU")
+    @PersistenceContext(unitName = "VDCNet-ejbPU-1")
     protected EntityManager em;
+    
+    @PersistenceContext(unitName = "VDCNet-ejbPU-2")
+    protected EntityManager em2;
+    
+    public EntityManager chooseEM(User user) {
+        if(user.isAuthenticated())  {
+            return em;
+        }
+        return em2;
+    }
 
     public Dataset find(Object pk) {
-        return em.find(Dataset.class, pk);
+        return chooseEM(session.getUser()).find(Dataset.class, pk);
```

Comment:
getEntityManager is probably a better name.

Reply:
Fixed.